routing/probability_bimodal_test.go

```diff
@@ -242,6 +242,39 @@ func TestSuccessProbability(t *testing.T) {
 	})
 }
 
+// TestSmallScale tests that the probability formula works with small scale
+// values.
+func TestSmallScale(t *testing.T) {
+	var (
+		// We use the smallest possible scale value together with a
+		// large capacity. This is an extreme form of a bimodal
+		// distribution.
+		scale    lnwire.MilliSatoshi = 1
+		capacity lnwire.MilliSatoshi = 7e+09
+
+		// Success and failure amounts are chosen such that the expected
+		// balance must be somewhere in the middle of the channel, a
+		// value not expected when dealing with a bimodal distribution.
+		// In this case, the bimodal model fails to give good forecasts
+		// due to the numerics of the exponential functions, which get
+		// evaluated to exact zero floats.
+		successAmount lnwire.MilliSatoshi = 1.0e+09
+		failAmount    lnwire.MilliSatoshi = 4.0e+09
+	)
+
+	estimator := BimodalEstimator{
+		BimodalConfig: BimodalConfig{BimodalScaleMsat: scale},
+	}
+
+	// An amount that's close to the success amount should have a very high
+	// probability.
+	amtCloseSuccess := successAmount + 1
+	_, err := estimator.probabilityFormula(
+		capacity, successAmount, failAmount, amtCloseSuccess,
+	)
+	require.ErrorContains(t, err, "normalization factor is zero")
```

Comment:
Interesting that in only fails in the reNorm calculation not already when calculating the prob:

```
	prob := p.integral(capacity, amount, failAmount)

```

Reply:
I think both calculations for `prob` and `reNorm` don't really "fail", but they return float64(0), not some non-zero small amount, because
```
	ecs := math.Exp(-c / s)
	exs := math.Exp(-x / s)
```
both give `float64(0)`, because `c/s` and `x/s` being very large. So in that sense both calculations are almost correct, but due to the division by plain zero we fail.